### PR TITLE
refactor(add): share workspace mutation orchestration

### LINKED ISSUE #693: refactor(add): extract shared workspace mutation orchestration for add-kind scaffolds
## Summary

Add-kind scaffolds should share common workspace mutation orchestration instead of repeating snapshot/patch/write/inventory/rollback flows.

## Problem

Complex add kinds such as `admin-view`, `ability`, and `ai-feature` follow the same orchestration pattern:

1. capture a workspace mutation snapshot;
2. patch PHP/bootstrap anchors;
3. patch package scripts/dependencies;
4. write generated files;
5. update workspace inventory;
6. roll back on failure.

Each kind currently owns much of this orchestration. New add kinds can easily duplicate hundreds of lines of boilerplate.

## Proposal

Create a shared workspace mutation executor or plan API for add-kind scaffolds.

Example direction:

```ts
interface WorkspaceMutationPlan {
  bootstrapAnchors?: unknown[];
  packagePatches?: unknown[];
  generatedFiles?: unknown[];
  inventoryEntries?: unknown[];
}
```

The exact shape can evolve, but common snapshot and rollback behavior should be centralized.

## Acceptance criteria

- [ ] Introduce a shared workspace mutation executor or plan abstraction.
- [ ] At least two complex add kinds use the shared mutation path.
- [ ] Shared PHP snippet insert/append helpers replace local duplicates where practical.
- [ ] Rollback behavior remains equivalent or improves.
- [ ] Existing add dry-run behavior remains intact.
- [ ] Tests cover success and rollback paths for the shared orchestration.

### PR DESCRIPTION
**Pull request by imjlk**
## Summary

- Add a shared workspace mutation executor that snapshots files and rolls back target paths on scaffold failures.
- Move ability, AI feature, and admin view scaffolds onto the shared mutation path.
- Share PHP snippet insertion helpers used by ability/admin-view bootstrap patching.
- Add focused success/rollback tests plus boundary coverage that keeps complex add kinds on the shared executor.

Fixes #693.

## Validation

- bun run --filter @wp-typia/project-tools build
- bun test packages/wp-typia-project-tools/tests/cli-add-workspace-mutation.test.ts packages/wp-typia-project-tools/tests/cli-add-workspace-boundaries.test.ts
- bun run --filter @wp-typia/project-tools test:workspace
- bun run typecheck
- node scripts/check-repo-format.mjs
- node scripts/validate-sampo-changesets.mjs
- git diff --check

## Release Notes

* **Refactor**
  * Improved workspace scaffolding reliability through centralized mutation orchestration with automatic rollback capabilities when operations fail.
  * Standardized PHP code insertion across workspace generation, ensuring consistent behavior across all scaffold types.

### CODE REVIEW

**Comment by imjlk** [comment]
@codex review

File: packages/wp-typia-project-tools/src/runtime/cli-add-workspace-mutation.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		snapshotDirs: [...snapshotDirs],
+		targetPaths: [...targetPaths],
+	};
+
+	try {
+		return await run();
+	} catch (error) {
+		await rollbackWorkspaceMutation(mutationSnapshot);
+		throw error;
+	}
```
[the remark this answers is not in the document]
Comment (imjlk) [reply packages/wp-typia-project-tools/src/runtime/cli-add-workspace-mutation.ts:61]: Patched in e9e0d94d: executeWorkspaceMutationPlan now throws WorkspaceMutationRollbackError when rollback also fails, preserving both mutationError and rollbackError. Added a regression test that forces rollback restoration to fail and asserts both errors remain available.

**Comment by imjlk** [comment]
@codex review
